src/stapi_fastapi/routers/product_router.py

```diff
@@ -178,13 +178,10 @@ async def search_opportunities(
             request,
         ):
             case Success((features, Some(pagination_token))):
-                links.append(self.order_link(request))
-                search.next = pagination_token
-                links.append(
-                    self.pagination_link(request, search.model_dump(mode="json"))
-                )
+                links.append(self.order_link(request, search))
+                links.append(self.pagination_link(request, search, pagination_token))
```

Comment:
this pushes the logic for how to construct the link down into the `pagination_link` method